scipy/interpolate/_interpolate.py

```diff
@@ -93,11 +93,42 @@ def lagrange(x, w):
 # !! found, get rid of it!
 
 
+dep_mesg = """\
+`interp2d` is deprecated in SciPy 1.10 and will be removed in SciPy 1.12.0. 
+
+For legacy code, nearly bug-for-bug compatible replacements are
+`RectBivariateSpline` on regular grids, and `bisplrep`/`bisplev` for
+scattered 2D data.
+
+In new code, for regular grids use `RegularGridInterpolator` instead.
+For scattered data, prefer `LinearNDInterpolator` or
+`CloughTocher2DInterpolator`.
+
+For more details see
+`https://gist.github.com/ev-br/8544371b40f414b7eaf3fe6217209bff`
+"""
+
 class interp2d:
     """
     interp2d(x, y, z, kind='linear', copy=True, bounds_error=False,
              fill_value=None)
 
+    .. deprecated:: 1.10.0
+
+    `interp2d` is deprecated in SciPy 1.10 and will be removed in SciPy 1.12.0. 
+
+    For legacy code, nearly bug-for-bug compatible replacements are
+    `RectBivariateSpline` on regular grids, and `bisplrep`/`bisplev` for
+    scattered 2D data.
+
+    In new code, for regular grids use `RegularGridInterpolator` instead.
+    For scattered data, prefer `LinearNDInterpolator` or
+    `CloughTocher2DInterpolator`.
+
+    For more details see
+    `https://gist.github.com/ev-br/8544371b40f414b7eaf3fe6217209bff`
```

Comment:
```suggestion
       `interp2d` is deprecated in SciPy 1.10 and will be removed in SciPy
       1.12.0. 

       For legacy code, nearly bug-for-bug compatible replacements are
       `RectBivariateSpline` on regular grids, and `bisplrep`/`bisplev` for
       scattered 2D data.

       In new code, for regular grids use `RegularGridInterpolator` instead.
       For scattered data, prefer `LinearNDInterpolator` or
       `CloughTocher2DInterpolator`.

       For more details see
       `https://gist.github.com/ev-br/8544371b40f414b7eaf3fe6217209bff`
```

I think this needs to be indented to show up within the directive.
Before:
![image](https://user-images.githubusercontent.com/60778417/195790354-0fbaa481-06a0-43e5-8876-6aa0cb169f72.png)
After:
![image](https://user-images.githubusercontent.com/60778417/195790393-d19d0d49-14dd-4b6d-902d-cad5b3141d9a.png)

Reply:
Cool, I did not know about that!